cmd/kpod/create.go

```diff
@@ -0,0 +1,899 @@
+	capAdd         []string // cap-add
+	capDrop        []string // cap-drop
+	cidFile        string
+	cgroupParent   string // cgroup-parent
+	command        []string
+	detach         bool         // detach
+	devices        []*pb.Device // device
+	dnsOpt         []string     //dns-opt
+	dnsSearch      []string     //dns-search
+	dnsServers     []string     //dns
+	entrypoint     string       //entrypoint
+	env            []string     //env
+	expose         []string     //expose
+	groupAdd       []uint32     // group-add
+	hostname       string       //hostname
+	image          string
+	interactive    bool              //interactive
+	ip6Address     string            //ipv6
+	ipAddress      string            //ip
+	labels         map[string]string //label
+	linkLocalIP    []string          // link-local-ip
+	logDriver      string            // log-driver
+	logDriverOpt   []string          // log-opt
+	macAddress     string            //mac-address
+	name           string            //name
+	network        string            //network
+	networkAlias   []string          //network-alias
+	nsIPC          string            // ipc
+	nsNet          string            //net
+	nsPID          string            //pid
+	nsUser         string
+	pod            string   //pod
+	privileged     bool     //privileged
+	publish        []string //publish
+	publishAll     bool     //publish-all
+	readOnlyRootfs bool     //read-only
+	resources      createResourceConfig
+	rm             bool              //rm
+	securityOpts   []string          //security-opt
+	sigProxy       bool              //sig-proxy
+	stopSignal     string            // stop-signal
+	stopTimeout    int64             // stop-timeout
+	storageOpts    []string          //storage-opt
+	sysctl         map[string]string //sysctl
+	tmpfs          []string          // tmpfs
+	tty            bool              //tty
+	user           uint32            //user
+	group          uint32            // group
+	volumes        []string          //volume
+	volumesFrom    []string          //volumes-from
+	workDir        string            //workdir
+}
+
+var createDescription = "Creates a new container from the given image or" +
+	" storage and prepares it for running the specified command. The" +
+	" container ID is then printed to stdout. You can then start it at" +
+	" any time with the kpod start <container_id> command. The container" +
+	" will be created with the initial state 'created'."
+
+var createCommand = cli.Command{
+	Name:        "create",
+	Usage:       "create but do not start a container",
+	Description: createDescription,
+	Flags:       createFlags,
+	Action:      createCmd,
+	ArgsUsage:   "IMAGE [COMMAND [ARG...]]",
+}
+
+func createCmd(c *cli.Context) error {
+	// TODO should allow user to create based off a directory on the host not just image
+	// Need CLI support for this
+	if err := validateFlags(c, createFlags); err != nil {
+		return err
+	}
+	//runtime, err := getRuntime(c)
```

Comment:
Why aren't we using getRuntime() here? As it is, this won't override the default c/storage configuration with the one in the config file (or passed as a flag)